mesonbuild/interpreterbase/decorators.py

```diff
@@ -425,7 +422,6 @@ def evolve(self, *,
                deprecated: T.Union[str, None, _NULL_T] = _NULL,
                deprecated_message: T.Union[str, None, _NULL_T] = _NULL,
                deprecated_values: T.Union[T.Dict[T.Union[_T, T.Type[T.List], T.Type[T.Dict]], T.Union[str, T.Tuple[str, str]]], None, _NULL_T] = _NULL,
```

Comment:
```suggestion
               deprecated_values: T.Union[T.Dict[T.Union[_T, type, ContainerTypeInfo], T.Union[str, T.Tuple[str, str]]], None, _NULL_T] = _NULL,
```

And the same for `since_values` above (which github won't let me comment on for some reason)

Reply:
done